src/ziggurat/kafka_consumer/consumer_handler.clj

```diff
@@ -42,15 +48,18 @@
         (let [start-time             (Instant/now)
               result                 (batch-handler batch)
               time-taken-in-millis   (.toMillis (Duration/between start-time (Instant/now)))]
-          (when (map? result)
-            (let [messages-to-be-retried (:retry result)
-                  to-be-retried-count    (count messages-to-be-retried)
-                  skip-count             (count (:skip result))
-                  success-count          (- batch-size (+ to-be-retried-count skip-count))]
-              (log/infof "[Consumer Group: %s] Processed the batch with success: [%d], skip: [%d] and retries: [%d] \n"
-                         topic-entity success-count skip-count to-be-retried-count)
-              (publish-batch-process-metrics topic-entity batch-size success-count skip-count to-be-retried-count time-taken-in-millis)
-              (retry messages-to-be-retried current-retry-count topic-entity)))))
+          (when-not (validate-return-type result)
```

Comment:
why not use clojure's pre conditions? :pre (or :post) on the function definition